src/libcore/iter/mod.rs

```diff
@@ -957,6 +967,29 @@ impl<A, B> DoubleEndedIterator for Chain<A, B> where
         }
     }
 
+    #[inline]
+    fn nth_back(&mut self, mut n: usize) -> Option<A::Item> {
+        match self.state {
+            ChainState::Both | ChainState::Front => {
+                for x in self.a.by_ref().rev() {
```

Comment:
This should be checking `Both | Back` first, pulling from `self.b`, and then `Front => self.a`.